ion/symboltoken.go

```diff
@@ -89,22 +92,75 @@ func (st *SymbolToken) Equal(o *SymbolToken) bool {
 	return false
 }
 
+// Parses text of the form '$n' for some integer n.
```

Comment:
I think I remember seeing similar code in #149, which has been merged. We should consolidate these methods later if they're fulfilling the same purpose.

Reply:
That similar code has already been consolidated in this PR.